frontend/apps/reader/skimtowidget.lua

```diff
@@ -58,63 +59,72 @@ function SkimToWidget:init()
             },
          }
     end
+    local dialog_title
     if self.document.info.has_pages then
-        self.dialog_title = _("Go to Page")
+        dialog_title = _("Go to Page")
         self.curr_page = self.ui.paging.current_page
     else
-        self.dialog_title = _("Go to Location")
+        dialog_title = _("Go to Location")
         self.curr_page = self.document:getCurrentPage()
     end
     self.page_count = self.document:getPageCount()
 
-    self.skimto_title = FrameContainer:new{
+    local ticks_candidates = {}
+    if self.ui.toc then
+        local max_level = self.ui.toc:getMaxDepth()
+        for i = 0, -max_level, -1 do
+            local ticks = self.ui.toc:getTocTicks(i)
+            table.insert(ticks_candidates, ticks)
+        end
+        -- find the finest toc ticks by sorting out the largest one
+        table.sort(ticks_candidates, function(a, b) return #a > #b end)
+    end
+    if #ticks_candidates > 0 then
+        self.ticks_candidates = ticks_candidates[1]
+    end
+
+    local skimto_title = FrameContainer:new{
         padding = Size.padding.default,
         margin = Size.margin.title,
         bordersize = 0,
         TextWidget:new{
-            text = self.dialog_title,
+            text = dialog_title,
             face = self.title_face,
             bold = true,
             width = self.screen_width * 0.95,
         },
     }
 
-    self.skimto_container = CenterContainer:new{
-        dimen = Geom:new{ w = self.screen_width * 0.95, h = self.screen_height * 0.075 },
-    }
-    local progress_bar = ProgressWidget:new{
+    self.progress_bar = ProgressWidget:new{
         width = self.screen_width * 0.9,
         height = Screen:scaleBySize(30),
         percentage = self.curr_page / self.page_count,
-        ticks = nil,
-        last = nil,
+        ticks = self.ticks_candidates,
+        tick_width = 1,
```

Comment:
Looks like the check is correct about this needing to be scaleBySized? Might consider doing Size.line.thin.